File: Fragment-III_TotalApple_Count_03Aug21/app/src/main/java/com/example/fragmentiiitotalapplecount03aug21/MainActivity.java
package com.example.fragmentiiitotalapplecount03aug21;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity implements CommunicationListener{
    private FragmentManager fragmentManager;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        fragmentManager = getSupportFragmentManager();
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        TotalApplesFragment totalApplesFragment = new TotalApplesFragment();
        transaction.add(R.id.container, totalApplesFragment, "totalAppleFragment").commit();
    }

    @Override
    public void LaunchBuyApplesFragment(Bundle bundle) {
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        BuyApplesFragment buyApplesFragment = new BuyApplesFragment();
        buyApplesFragment.setArguments(bundle);
        transaction.replace(R.id.container, buyApplesFragment, "buyFragment").addToBackStack("buyFragment").commit();
    }

    @Override
    public void LaunchTotalApplesFragment(Bundle bundle) {
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        TotalApplesFragment totalApplesFragment = new TotalApplesFragment();
        totalApplesFragment.setArguments(bundle);
        transaction.replace(R.id.container, totalApplesFragment, "totalAppleFragment").addToBackStack("totalAppleFragment").commit();
    }


}

File: _Kotlin Projects/Kotlin_Scoping_Functions_IW_19Aug21/app/src/main/java/com/example/kotlin_scoping_functions_iw_19aug21/MainActivity.kt
package com.example.kotlin_scoping_functions_iw_19aug21

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.FragmentManager

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        launchFragment()
    }

    private fun launchFragment() {
        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.add(R.id.container, UserDetailFragment(), "UserDetailFragment").commit()
    }
}